libtransmission/tr-lpd.h

```diff
@@ -8,9 +8,47 @@
 #error only libtransmission should #include this header.
 #endif
 
-int tr_lpdInit(tr_session*, tr_address*);
-void tr_lpdUninit(tr_session*);
-bool tr_lpdSendAnnounce(tr_torrent const*);
+#include <ctime>
+#include <memory>
+#include <string_view>
+#include <vector>
 
-/**
-* @} */
+#include "transmission.h"
+
+#include "net.h" // for tr_address, tr_port
+#include "timer.h"
+
+class tr_torrents;
+struct tr_session;
+struct event_base;
+
+class tr_lpd
```

Comment:
nit: I think `tr_lpd` itself can be final without virtual dtor. At least I do not see any inherited classes.

Reply:
Or maybe move static factory func in Mediator itself and tr_lpd its not needed

Reply:
I think I'm misunderstanding this comment? `tr_lpd` is a pure virtual class; the inherited impl class is `tr_lpd_impl` in `libtransmission/tr-lpd.cc`

Reply:
> I think I'm misunderstanding this comment?

Nope, you are right.

As I said 

> At least I do not see any inherited classes.

tr-lpd.cc haven't loaded for me as diff is too large for GitHub's defaults.